spark-hpcc/src/main/java/org/hpccsystems/spark/HpccFile.java

```diff
@@ -166,6 +168,35 @@ public HpccRDD getRDD(SparkContext sc) throws HpccFileException
     rdd.setTraceContext(parentTraceID, parentSpanID);
     return rdd;
   }
+
+  /**
+   * Make a Spark Resilient Distributed Dataset (RDD) that provides access
+   * to THOR based datasets.
+   * @param sc Spark Context
+   * @return An RDD of THOR data.
+   * @throws HpccFileException When there are errors reaching the THOR data
+   */
+  public HpccRDD getRDD(SparkContext sc, List<Integer> filePartsToRead) throws HpccFileException
```

Comment:
have we considered breaking out early if the List is empty?
Also, are there other situations we should avoid such as reading the same part multiple times?

Reply:
Good question, will make the change to check for empty file parts list. I think there are some valid cases where reading the same file multiple times across the cluster is desired.

Reply:
fair enough, but what are those cases, and how can we document them?